### PR TITLE
Fix some book typos

### PR DESCRIPTION
**Pull request by spastorino**
I think these changes are correct but check it out just in case, there's one or two that I'm not 100% sure.

### CODE REVIEW

File: book/src/clauses/implied_bounds.md [hunk 1]
```diff
@@ -183,7 +183,7 @@ of  "local" implied bound reasoning which would say
 only be done within our `foo` function in order to avoid the earlier
 problem where we had a global clause.
 
-We can apply these local reasonings everywhere we can have an environment
+We can apply these local reasoning everywhere we can have an environment
```
Comment (detrumi) [inline book/src/clauses/implied_bounds.md:186]: Either "these local reasonings" or "this local reasoning"

File: book/src/clauses/coherence.md [hunk 2]
```diff
@@ -166,7 +166,7 @@ The conclusion from all of this is that it is perfectly safe to rule out impls t
 
 **Downstream Crates:** Downstream crates come into play because all traits in upstream crates and in the current crate can potentially be implemented by downstream crates using the forms allowed by the orphan rules. In essence, we always need to assume that downstream crates will implement traits in all ways that compile.
 
-## Discussion: Modelling the Overlap Check
+## Discussion: Modeling the Overlap Check
```
Comment (detrumi) [inline book/src/clauses/coherence.md:169]: Looks like one or 2 L's are both fine, depending on whether it's American English or not. But I think we default to American English in most places, so :+1:

File: book/src/clauses/opaque_types.md [hunk 3]
```diff
@@ -75,7 +75,7 @@ A chalk opaque type declaration has several parts:
 * The **where clauses**, which would be `T: Copy` and `T: Into<u32>` in our
   example. These are conditions that must hold on `V0..Vn` for
   `OpaqueTypeName<V0..Vn>` to be a valid type.
-    * Note that constrast with bounds: bounds are things that the hidden type must meet
+    * Note that contrast with bounds: bounds are things that the hidden type must meet
```
Comment (detrumi) [inline book/src/clauses/opaque_types.md:78]: Maybe this should be "Note the contrast"?

File: book/src/engine/slg.md [hunk 4]
```diff
@@ -30,7 +30,7 @@ done. This is similar to how Prolog works.
 `Vec<?T>: Debug` is true if `?T: Debug`. This leads to a cycle: `[Vec<u32>,
 Vec<Vec<u32>>, Vec<Vec<Vec<u32>>>]`, and so on all implement `Debug`. Our
 solver ought to be breadth first and consider answers like `[Vec<u32>: Debug,
-Vec<i32>: Debug, ...]` before it recurses, or we may never find the answer
+Vec<i32>: Debug, ...]` before it recourses, or we may never find the answer
```
Comment (detrumi) [inline book/src/engine/slg.md:33]: I think this was using "recursion" as verb here, so it was correct?

File: book/src/types/rust_types.md [hunk 5]
```diff
@@ -142,7 +142,7 @@ them in the [aliases chapter](./rust_types/alias.md).
 The `BoundVar` variant represents some variable that is bound in
 an outer term. For example, given a term like `forall<X> {
 Implemented(X: Trait) }`, the `X` is bound. Bound variables in chalk
-(like rustc) use de bruijin indices (See below).
+(like rustc) use de bruijn indices (See below).
```
Comment (detrumi) [inline book/src/types/rust_types.md:145]: We should change this to "De Bruijn indices"

**Review by detrumi — CHANGES_REQUESTED** [review]

File: book/src/clauses/opaque_types.md (hunk 3 above)
Comment (detrumi) [reply book/src/clauses/opaque_types.md:78]: Or the intended text might've been "Note that this contrasts..."

File: book/src/clauses/coherence.md (hunk 2 above)
Comment (spastorino) [reply book/src/clauses/coherence.md:169]: Oh I had no idea that it was correct in non American English. I can leave it as it was.

File: book/src/types/rust_types.md (hunk 5 above)
Comment (spastorino) [reply book/src/types/rust_types.md]: Changed all occurrences to De Bruijn

**Review by detrumi — APPROVED** [review]

**Pull request merged**